certinel.go

```diff
@@ -65,15 +64,6 @@ func (c *Certinel) Close() error {
 // can be passed as the GetCertificate member in a tls.Config object. It is
 // safe to call across multiple goroutines.
 func (c *Certinel) GetCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
-
-	return c.certificate, nil
-}
-
-func (c *Certinel) loadCertificate(certificate tls.Certificate) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
-	c.certificate = &certificate
+	cert, _ := c.certificate.Load().(*tls.Certificate)
```

Comment:
```go
package main

import (
	"crypto/tls"
	"errors"
	"log"
	"net/http"
)

var (
	errNoCertificate = errors.New("error!")
)

func main() {
	server := http.Server{
		Addr: ":8000",
		TLSConfig: &tls.Config{
			GetCertificate: func(hello *tls.ClientHelloInfo) (*tls.Certificate, error) {
				// return nil, nil
				return nil, errNoCertificate
			},
		},
	}

	log.Printf("info: listening at https://localhost:8000/")
	server.ListenAndServeTLS("", "")
}
```

Sends an internal error alert to the client (to close the connection) and prints a log line.

```
$ http https://localhost:8000

http: error: SSLError: [SSL: TLSV1_ALERT_INTERNAL_ERROR] tlsv1 alert internal error (_ssl.c:720) while doing GET request to URL: https://localhost:8000/
```

```
2017/05/26 07:57:57 http: TLS handshake error from [::1]:54641: error!
```

Returning (nil, nil) has the same behavior, just with a different error in the logs

```
2017/05/26 07:59:03 http: TLS handshake error from [::1]:54846: tls: no certificates configured
```